tooling/nargo_cli/src/cli/compile_cmd.rs

```diff
@@ -227,52 +222,45 @@ fn save_program(
     }
 }
 
-fn save_contracts(
-    contracts: Vec<(CompiledContract, DebugArtifact)>,
+fn save_contract(
+    contract: (CompiledContract, DebugArtifact),
     package: &Package,
     circuit_dir: &Path,
     output_debug: bool,
 ) {
+    let (contract, debug_artifact) = contract;
+
     // TODO(#1389): I wonder if it is incorrect for nargo-core to know anything about contracts.
     // As can be seen here, It seems like a leaky abstraction where ContractFunctions (essentially CompiledPrograms)
     // are compiled via nargo-core and then the PreprocessedContract is constructed here.
     // This is due to EACH function needing it's own CRS, PKey, and VKey from the backend.
```

Comment:
orthogonal: oof this seems outdated

Reply:
Just waiting on https://github.com/noir-lang/noir/pull/2666